### PR TITLE
Add Support for ServerSideEncryption including using KMS key Id

### PR DESCRIPTION
**Pull request by pmmenzel**
Add AWS S3 server side encryption attributes to `CreateS3` and `UpdateS3` functions and update any associated tests.

Closes https://github.com/fastly/go-fastly/issues/141

### CODE REVIEW

File: fastly/s3.go [hunk 1]
```diff
@@ -116,6 +123,10 @@ func (c *Client) CreateS3(i *CreateS3Input) (*S3, error) {
 		return nil, ErrMissingVersion
 	}
 
+	if i.ServerSideEncryption == S3ServerSideEncryptionKMS && i.ServerSideEncryptionKmsKeyId == "" {
+		return nil, ErrMissingKmsKeyId
+	}
```
Comment (philippschulte) [inline fastly/s3.go]: Nice!

File: fastly/s3_test.go [hunk 2]
```diff
@@ -228,6 +243,17 @@ func TestClient_CreateS3_validation(t *testing.T) {
...
+	_, err = testClient.CreateS3(&CreateS3Input{
+		Service:                      "foo",
+		Version:                      1,
+		Name:                         "test-service",
+		ServerSideEncryption:         S3ServerSideEncryptionKMS,
+		ServerSideEncryptionKmsKeyId: "",
+	})
+	if err != ErrMissingKmsKeyId {
+		t.Errorf("bad error: %s", err)
+	}
```
Comment (philippschulte) [inline fastly/s3_test.go]: Nice!

File: fastly/fixtures/s3s/create.yaml [hunk 3]
```diff
@@ -47,7 +47,7 @@ interactions:
...
+    body: '{"access_key":"AKIAIOSFODNN7EXAMPLE","bucket_name":"bucket-name","domain":"s3.us-east-1.amazonaws.com","format":"format","format_version":"2","gzip_level":9,"message_type":"classic","name":"test-s3","path":"/path","period":12,"placement":"waf_debug","redundancy":"reduced_redundancy","secret_key":"wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY","timestamp_format":"%Y","service_id":"7i6HN3TK9wS159v2gPAZ8A","version":"41","server_side_encryption":"aws:kms","server_side_encryption_kms_key_id":"1234","response_condition":"","updated_at":"2019-05-09T22:30:57Z","acl":null,"deleted_at":null,"public_key":null,"created_at":"2019-05-09T22:30:57Z"}'
```
[the remark this answers is not in the document]
Comment (pmmenzel) [reply fastly/fixtures/s3s/create.yaml]: @philippschulte I followed the instructions in the link you provided.  Specifically, I'm running `go test -v -timeout 30s -run TestClient_S3s`.   However, I am getting the following error/failure:
```
fastly_test.go:92: Post https://api.fastly.com/service/7KZSzyXYPlSON4svYqbmDz/version: Requested interaction not found
```
This relates to the `testVersion` function in `fastly_test.go`

Digging a little deeper, I was able to track this down to the [Request](https://github.com/fastly/go-fastly/blob/master/fastly/client.go#L211) method in `client.go`.   More specifically, it appears to be related to the `c.HTTPClient.Do(req)` being passed to `checkResp`.   This is where I get stuck as I'm not proficient in golang. 

Any guidance or help you could provide would be greatly appreciated.

[the remark this answers is not in the document]
Comment (pmmenzel) [reply fastly/fixtures/s3s/create.yaml]: @philippschulte  Nevermind.   I removed all the recordings and the tests passed just fine ✅ !

I think that part wasn't clear to me in the instructions.

File: fastly/s3.go [hunk 4]
```diff
@@ -8,35 +8,40 @@ import (
...
+	ResponseCondition            string                 `mapstructure:"response_condition"`
+	MessageType                  string                 `mapstructure:"message_type"`
+	TimestampFormat              string                 `mapstructure:"timestamp_format"`
+	Placement                    string                 `mapstructure:"placement"`
+	Redundancy                   S3Redundancy           `mapstructure:"redundancy"`
+	CreatedAt                    *time.Time             `mapstructure:"created_at"`
+	UpdatedAt                    *time.Time             `mapstructure:"updated_at"`
+	DeletedAt                    *time.Time             `mapstructure:"deleted_at"`
+	ServerSideEncryptionKMSKeyID string                 `mapstructure:"server_side_encryption_kms_key_id"`
+	ServerSideEncryption         S3ServerSideEncryption `mapstructure:"server_side_encryption"`
```
Comment (phamann) [inline fastly/s3.go]: Minor nit: please place these above the `CreatedAt` etc metadata fields to follow convention with the rest of the project.

**Pull request merged**